### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by NikitaKuzminn**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,4 +9,8 @@
 
 # Maven
 log/
-target/
\ No newline at end of file
+target/
+
+#Allure
+.allure
+allure-results/
```
Comment (komyagin) [inline .gitignore:16]: Add a new blank line.

File: pom.xml [hunk 2]
```diff
@@ -105,6 +117,14 @@
       <systemPath>${project.basedir}/src/main/resources/calculator-1.0.jar</systemPath>
     </dependency>
```
Comment (komyagin) [inline pom.xml:119]: Remove unnecessary blank line

File: src/test/java/home/epam/hw3/confing/ConfProperties.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    public ConfProperties() {
+    }
+
+    public static String getProperty(String key) {
+        return PROPERTIES.getProperty(key);
+    }
+
+    static {
+        try {
+            fileInputStream = new FileInputStream("src/test/resources/home.epam.hw3/conf.properties");
```
Comment (komyagin) [inline src/test/java/home/epam/hw3/confing/ConfProperties.java:20]: Move string to the string constant

File: src/test/java/home/epam/hw3/confing/BaseTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import io.github.bonigarcia.wdm.WebDriverManager;
+import java.util.concurrent.TimeUnit;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+
+public class BaseTest {
+    protected WebDriver webDriver;
+    protected String pageID;
```
Comment (komyagin) [inline src/test/java/home/epam/hw3/confing/BaseTest.java:12]: What this stands for?

File: src/test/java/home/epam/hw3/pages/DifferentElementsPage.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    @FindBy(xpath = "//label[@class=\"label-radio\"]")
+    private List<WebElement> radio;
+
+    @FindBy(className = "colors")
+    private WebElement colors;
+
+    @FindBy(xpath = "//option[text()=\"Yellow\"]")
+    private WebElement color;
+
+    @FindBy(css = ".info-panel-body-log .panel-body-list > li")
```
Comment (komyagin) [inline src/test/java/home/epam/hw3/pages/DifferentElementsPage.java:28]: Are there any other options?

File: src/test/java/home/epam/hw3/pages/HomePage.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        webDriver.navigate().to(url);
+    }
+
+    public String getPageTitle() {
+        return webDriver.getTitle();
+    }
+
+    public void switchOn(boolean flag) {
+        webDriver = (flag) ? this.webDriver.switchTo().frame(buttonFrame) : this.webDriver.switchTo().defaultContent();
+    }
```
Comment (komyagin) [inline src/test/java/home/epam/hw3/pages/HomePage.java:42]: What this stands for?

**Review by komyagin — CHANGES_REQUESTED** [review]

File: src/test/java/home/epam/hw3/pages/HomePage.java (hunk 6 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw3/pages/HomePage.java]: Fixed

File: src/test/java/home/epam/hw3/pages/HeaderMenu.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        loginIcon.click();
+        loginField.sendKeys(ConfProperties.getProperty("login"));
+        passField.sendKeys(ConfProperties.getProperty("password"));
+        loginButton.click();
+    }
+
+    public void logout() {
+        try {
+            logoutButton.click();
+        } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw3/pages/HeaderMenu.java]: Fixed

File: src/test/java/home/epam/hw3/pages/HeaderMenu.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+    public HeaderMenu(WebDriver webDriver) {
+        super(webDriver);
+        PageFactory.initElements(webDriver, this);
+    }
+
+    public void login() {
+        loginIcon.click();
+        loginField.sendKeys(ConfProperties.getProperty("login"));
+        passField.sendKeys(ConfProperties.getProperty("password"));
```
[the remark this answers is not in the document]
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw3/pages/HeaderMenu.java]: Fixed

File: src/test/java/home/epam/hw3/pages/DifferentElementsPage.java (hunk 5 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw3/pages/DifferentElementsPage.java]: Fixed

File: src/test/java/home/epam/hw3/confing/BaseTest.java (hunk 4 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw3/confing/BaseTest.java]: Fixed

File: src/test/java/home/epam/hw3/confing/ConfProperties.java (hunk 3 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw3/confing/ConfProperties.java]: Fixed

File: pom.xml [hunk 9]
```diff
@@ -105,6 +117,14 @@
...
 
+    <dependency>
+      <groupId>org.projectlombok</groupId>
+      <artifactId>lombok</artifactId>
+      <version>RELEASE</version>
+      <scope>test</scope>
+    </dependency>
+
+
```
[the remark this answers is not in the document]
Comment (NikitaKuzminn) [reply pom.xml]: Fixed

File: pom.xml (hunk 2 above)
Comment (NikitaKuzminn) [reply pom.xml:120]: Fixed

File: .gitignore (hunk 1 above)
Comment (NikitaKuzminn) [reply .gitignore]: Fixed

**Review by komyagin — APPROVED** [review]

**Pull request merged**